interview-questions.md

```diff
@@ -7,44 +7,56 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating an API with a model called Animal that has_many Sightings, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be part of the Animal model or the Sightings model?
 
-  Your answer:
+  Your answer: The foreign key would be part of the Animal model and the name would be called animal_id. In order to fix this, you can go into the console and add a new migration.
 
-  Researched answer:
+  Researched answer: There must be a row already existing in order to add the foreign key.
 
 
 
 2. Which RESTful API routes must always be passed params? Why?
 
-  Your answer:
+  Your answer: show, update, destroy must always be passed params so the developer knows which item they are modifying or showing.
 
-  Researched answer:
+  Researched answer: 
+API parameters are the variable parts of a resource. They determine the type of action you want to take on the resource. Each parameter has a name, value type ad optional description. Whenever you want to build a REST API, you have to decide which parameters should be present in the API endpoint.
 
 
 
 3. Name three rails generator commands. What is created by each?
 
-  Your answer:
+  Your answer: rails g model, rails g resource, and rails g migration
 
-  Researched answer:
+  Researched answer: rails g model creates a new model with specific keys, rails g resource creates the model, migrations routes and controllers, and rails g migration runs a migration for controllers.
```

Comment:
yea, close on that last one $ rails g migration creates a migration file but $ rails db:migrate runs the migrations